src/main/java/org/littleshoot/proxy/impl/ClientToProxyConnection.java

```diff
@@ -1015,7 +1016,20 @@ private boolean authenticationRequired(HttpRequest request) {
         
         String userName = StringUtils.substringBefore(decodedString, ":");
         String password = StringUtils.substringAfter(decodedString, ":");
+
+        RateLimiter rateLimiter = proxyServer.getRateLimiter();
+
+        if(rateLimiter != null && rateLimiter.isAuthenticationOverLimit(userName)) {
```

Comment:
instead of null check can we have a `NoOpRateLimiter` by default that always returns false?

Reply:
the authenticator is never null so it would fit this pattern.

Reply:
Authenticator can be null for reverse proxy. I was trying to follow the same pattern that is used all over LittleProxy.